ex1/functions.py

```diff
@@ -66,23 +145,147 @@ def f():
+    n: int, float
+    
+    Returns
+    -------
+    List
+		Elements with True or False
+    
+    See Also
+    --------
+    check5(a)
+    check_n(a,n)
+    
+    Examples
+    --------
+    >>> print(check_list([1,2,5,10],3))
+    [False,False,True,True]
+    """
+    return [value>=n for value in numberlist]
+
 # Define a function named check_list_nth that does the same as check_list but
 # uses every nth element of the input list (including the first one). You will
 # need a third input argument.
 
+def check_list_nth(numberlist,n,nth):
+    """ Check if every nth element in list is greater or equal to n.
+    	
+    Parameters
+    ----------
+    numberlist: list
+		type of elements: number
+    n: int, float
+    nth: int
+    
+    Returns
+    -------
+    List
+		Elements with True or False
+    
+    See Also
+    --------
+    check5(a)
+    check_n(a,n)
+    check_list(numberlist,n)
+    
+    Examples
+    --------
+    >>> print(check_list_nth([1,2,10,4,5,6,7,10],5,3))
+    [False,False,True]
+    """
+    return [value>=n for value in numberlist[0::nth]]
+   
 # Define a function named add_new_list that takes two inputs. A list l and a
 # second variable x to add to the list. Return a new list containing x as the
 # last element without modifying the original list.
 
+def add_new_list(l,x):
+    """ Creates a copy of list l and adds the element x
+    	
+    Parameters
+    ----------
+    l: list
+    x: any
+    
+    Returns
+    -------
+    List
+    
+    Examples
+    --------
+    >>>l = [1,2,3,'tt']
+    >>>l2 =  add_new_list(l,3)
+    >>>print(l)
+    [1,2,3,'tt']
+    >>>print(l2)
+    [1,2,3,'tt',3]
+    """
+    l_copy = list(l)
+    l_copy.append(x)
+    return l_copy
+
 # Define a function named remove_nth that takes a list and removes every nth
 # element (including the first one). Use a keyword named nth to set the default
 # value for nth to 2.
 
+def remove_nth(l,nth=2):
+    """ Deletes every nth element in list l
+    	
+    Parameters
+    ----------
+    l: list
+    nth: int, optional
+    
+    Returns
+    -------
+    List
+    
+    Examples
+    --------
+    >>>print(remove_nth([1,2,3,4,5,6],3))
+    [2,3,5,6]
+    """
+    l_copy = list(l)
+    del l_copy[::nth]
+    return l_copy
+
 # Define a function named search_n that takes a list and a variable x and
 # searches for x in the list. If the variable is found return the index of the
 # variable in the list and the variable. Otherwise use None for both return
 # values
 
+def search_n(l,x):
+    """ Search for the first x in list l and returns a tuple with index and value. It only returns the first found value.
+    	
+    Parameters
+    ----------
+    l: list
+    x: any
+    
+    Returns
+    -------
+    List
+    
+    Examples
+    --------
+    >>>print(search_n([1,2,3,4,5,3,6],3))
+    (2,3)
+    """
+    erg = (None,None)
+    for idx,value in enumerate(l):
+        if value == x:
+            erg = (idx,value)
+            break
```

Comment:
This can be done easier. See the example solution at https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20